src/components/dashboard/main-content/course-enrollments/course-cards/CompletedCourseCard.jsx

```diff
@@ -1,36 +1,142 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
 import { getAuthenticatedUser } from '@edx/frontend-platform/auth';
+import { sendTrackEvent } from '@edx/frontend-platform/analytics';
 
 import BaseCourseCard from './BaseCourseCard';
+import ContinueLearningButton from './ContinueLearningButton';
+import { UnarchiveModal } from './unarchive-modal';
 
+import {
+  updateCourseRunStatus,
+  updateIsUnarchiveCourseSuccess,
+} from '../data/actions';
+import { isCourseEnded } from '../../../../../utils/common';
 import CertificateImg from './images/edx-verified-mini-cert.png';
 
-const CompletedCourseCard = props => {
+const CompletedCourseCard = (props) => {
   const user = getAuthenticatedUser();
   const { username } = user;
-  return (
-    <BaseCourseCard type="completed" hasViewCertificateLink={false} {...props}>
-      {props.linkToCertificate ? (
-        <div className="d-flex mb-3">
-          <div className="mr-3">
-            <img src={CertificateImg} alt="verified certificate preview" />
-          </div>
-          <div className="d-flex align-items-center">
-            <p className="lead mb-0 font-weight-normal">
-              View your certificate on
-              {' '}
-              <a className="text-underline" href={`${process.env.LMS_BASE_URL}/u/${username}`}>your profile →</a>
-            </p>
+  const {
+    markedDone,
+    title,
+    linkToCourse,
+    courseRunId,
+    courseRunStatus,
+    modifyCourseRunStatus,
+    modifyIsUnarchiveCourseStatus,
+    endDate,
+  } = props;
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const handleUnarchiveOnClose = () => {
+    setIsModalOpen(false);
+    sendTrackEvent('edx.learner_portal.course.unarchive.modal.closed', {
+      course_run_id: courseRunId,
+    });
+  };
+
+  const handleUnarchiveOnSuccess = ({ response, resetModalState }) => {
+    sendTrackEvent('edx.learner_portal.course.unarchive.saved', {
+      course_run_id: courseRunId,
+    });
+    setIsModalOpen(false);
+    resetModalState();
+    modifyCourseRunStatus({
+      status: response.courseRunStatus,
+      courseId: response.courseRunId,
+      markedDone: response.markedDone,
+    });
+    modifyIsUnarchiveCourseStatus({
+      isSuccess: true,
+    });
+  };
+
+  const getDropdownMenuItems = () => {
+    // Only courses that are manually archived (markedDone) should show a unarchive option.
+    // if course is ended or completed, you cannot unarchive it
+    // TODO: we also need to add || courseRunStatus === 'completed' once api returns correct status
```

Comment:
Just curious about the sequencing of the API fix to return correct status and this PR. Are you hoping to get this merged in, fix the API to return correct status, and then come back and update this repo?

Reply:
Yes, this part of the functionality does not actually work correctly until we fix the api issue. If I add this chekc now, it means we always hide the unarchive button. I would like to revisit this as soon as the api + this approach is finalized. I now think that using ENT-3028 as a pin to do that is a good idea, so I will leave that ticket around

Reply:
to be clear, it does work correctly but it fails to hide this button for courses that are actually complete.